enclave/src/aes.rs

```diff
@@ -15,66 +15,70 @@
 
 */
 
-use std::{sgxfs::SgxFile, vec::Vec};
-
-use sgx_rand::{Rng, StdRng};
-use sgx_types::*;
-
+use crate::error::{Error, Result};
 use aes::Aes128;
-use log::*;
+use codec::{Decode, Encode};
+use log::info;
 use ofb::{
 	cipher::{NewStreamCipher, SyncStreamCipher},
 	Ofb,
 };
+use sgx_rand::{Rng, StdRng};
+use std::{sgxfs::SgxFile, vec::Vec};
+use substratee_sgx_io::{seal, unseal, SealedIO};
 
-use crate::{io, utils::UnwrapOrSgxErrorUnexpected};
+use crate::utils::UnwrapOrSgxErrorUnexpected;
 use substratee_settings::files::AES_KEY_FILE_AND_INIT_V;
 
 type AesOfb = Ofb<Aes128>;
 
-pub type Aes = (Vec<u8>, Vec<u8>);
+#[derive(Debug, Default, Encode, Decode)]
+pub struct Aes {
+	pub key: [u8; 16],
```

Comment:
Would it make sense to define a `Type AesKey  = ...`  ( & same for `init_vec` )? In case we change the keytype due to whatever reason, it would make it easier to just adapt the code.

Reply:
This is good input. However, in the next PR I extracted the crypto crate and defined a trait `StateEncryption`, which allows even being agnostic about the key-type being used.